AirSkyBoat/AirSkyBoat
# maintain persistence of a player's merit mode selection

**_I affirm:_**


## Please enter a player-facing description

Maintains persistence of a player's merit mode selection though job changes

## What does this pull request do? (Please be technical)

This change removes a sql update call which was overwriting a player's meritMode setting.
Further more this change refactors the verification if a job should be allowed to be on merit mode into a new function and uses said function to validate and apply merit mode in the following cases:
- On Job Change
- On character load (zone, logging in, etc)
- On leveling up to level 75 specifically

## Steps to test these changes
stats addon helps as it shows different xp bars for xp and meritMode - so its a dead give away
Otherwise you have to rely on the blue char name in equip or status

get a char with the following:
- a level 74 or less job (lets say 60 for fun)
- a level 74 job
- a level 75 job wth uncapped xp
- a level 75 job with capped xp
- character must have genkai set to 75 or greater

USING A MOOGLE (do not use !changejob - it bypasses danm near everything)
Change to a level 75 job and set merit mode to EXP
change to the level 60 - note that you are not in merit mode
change to 74 -note you are not in merit mode
change to 75 uncapped xp - note you are not in merit mode
change to 75 capped xp - note you are in merit mode
Change merit mode to Limit Points (or whatever the menu says)
Change to the 75 uncapped - note that you are in merit mode
Change to the 60 - note that you are not in merit mode
Change back to 75 uncapped - note that you are in merit mode still (yay)
Change to 75 capped xp - note that you are in merit mode
Change to 74 - note that you are not in merit mode
use !givexp <number> to give enough xp to level up
Note that after leveling you now in merit mode
use the command !changejob <instert job that you just leveled up to 75> 74
Note you are still in merit mode - i mentioned that !changejob bypasses everything, didnt I?
Changejob to a 75 - note that you are in merit mode
Change merit mode to exp
Change to the 74
use !givexp <number> to level the job to 75
note that after leveling to 75 you are still in xp mode.

## Special Deployment Considerations

None
